libsql-wal/src/bottomless/storage/fs.rs

```diff
@@ -19,19 +30,60 @@ impl Storage for FsStorage {
         segment_data: impl crate::io::file::FileExt,
         segment_index: Vec<u8>,
     ) -> impl Future<Output = Result<()>> + Send {
-        todo!();
-        #[allow(unreachable_code)]
-        std::future::ready(Ok(()))
+        let key = format!(
+            "{}-{}-{}.segment",
```

Comment:
You want fixed length keys for two reasons:
- so that they are lexicographically sortable by start-frame 
- they are more efficiently splittable